Lib/unittest/test/test_program.py

```diff
@@ -289,6 +321,12 @@ def test_locals(self):
                                                'verbosity': 1,
                                                'warnings': None})
 
+    def test_debug(self):
+        program = self.program
+        program.testRunner = FakeRunner
+        program.parseArgs([None, '--debug'])
+        self.assertTrue(program.debug)
```

Comment:
I believe more tests are needed.

Reply:
Do you mean more tests for the argument parsing part? I've also added a general test that ensures that any exception is leaked for `--debug` rather than being consumed by unittest.

Reply:
@ZackerySpytz What kind of tests do you think should be added? I'm happy to add more, but at the moment it's not clear to me what exactly should be tested in addition to what's already there.

Reply:
Never mind.

Reply:
@ZackerySpytz I'm not familiar with the review process, what do I need to do to further advance this PR?